src/views/recordTable.component.test.tsx

```diff
@@ -110,106 +109,108 @@ describe('Record Table', () => {
   it('calls the correct data fetching hooks on load', () => {
     createView();
 
-    expect(useRecordsPaginated).toHaveBeenCalledWith({
-      page: 0,
-      sort: {},
-      dateRange: {},
-    });
+    expect(useRecordsPaginated).toHaveBeenCalled();
     expect(useRecordCount).toHaveBeenCalled();
     expect(useChannels).toHaveBeenCalled();
   });
 
-  it('updates page query parameter on page change', async () => {
-    data = Array.from(Array(12), (_, i) => generateRecord(i + 1));
-    (useRecordsPaginated as jest.Mock).mockReturnValue({
-      data: data,
-      isLoading: false,
-    });
-    (useRecordCount as jest.Mock).mockReturnValue({
-      data: data.length,
-      isLoading: false,
-    });
-    props.resultsPerPage = 10;
+  it('can sort columns and removes column sort when column is closed', async () => {
+    const user = userEvent.setup();
     createView();
 
+    await user.click(screen.getByLabelText('shotNum checkbox'));
+    await user.click(screen.getByTestId('sort shotNum'));
+
     await act(async () => {
-      screen.getByLabelText('Go to next page').click();
       await flushPromises();
     });
 
-    expect(useRecordsPaginated).toHaveBeenLastCalledWith({
-      page: 1,
-      sort: {},
-      dateRange: {},
-    });
-  });
+    expect(screen.getByTestId('sort shotNum')).toHaveClass('Mui-active');
 
-  it('updates sort query parameter on sort', async () => {
-    createView();
+    let menuIcon = screen.getByLabelText('shotNum menu');
+    fireEvent.click(menuIcon);
 
-    await act(async () => {
-      screen.getByTestId('sort timestamp').click();
-      await flushPromises();
-    });
+    const close = screen.getByText('Close');
+    fireEvent.click(close);
 
-    expect(useRecordsPaginated).toHaveBeenLastCalledWith({
-      page: 0,
-      sort: {
-        timestamp: 'asc',
-      },
-      dateRange: {},
-    });
+    expect(screen.getByLabelText('shotNum checkbox')).toHaveProperty(
+      'checked',
+      false
+    );
 
-    await act(async () => {
-      screen.getByTestId('sort timestamp').click();
-      await flushPromises();
-    });
+    await user.click(screen.getByLabelText('shotNum checkbox'));
 
-    expect(useRecordsPaginated).toHaveBeenLastCalledWith({
-      page: 0,
-      sort: {
-        timestamp: 'desc',
-      },
-      dateRange: {},
-    });
+    expect(screen.getByTestId('sort shotNum')).not.toHaveClass('Mui-active');
+  });
 
-    await act(async () => {
-      screen.getByTestId('sort timestamp').click();
-      await flushPromises();
+  // TODO: this test is similar to the one in table - either we keep the redux
+  // state in recordTable and do the test here - or we move the state into table
+  // and do the test there
```

Comment:
This kinda relates to my previous comment. What I'm a fan of is testing the correct methods are called and testing that state is correctly altered. This can be more accurate than testing for things on the UI like `"1-12 of ${data.length}"`

Reply:
So to answer your question, leaning towards moving state into table

Reply:
I thought the recommendation from `rtl` is to test the app how the user would, so the user would see that when the pagination is handled correctly that the UI text should change, rather than testing that specific Redux actions are sent & that state is updated, which are implementation details that the user can't see.

Reply:
Perhaps you're right. I just don't want to drift so far that we start testing MUI itself. I like testing units of functionality are doing the correct thing, while relying more on Cypress for what the user sees happening.

Anyway, I guess the main thing I wanna say on this particular comment is that, because there's already a fair bit of testing on pagination elsewhere, don't overdo it here. We know the correct page number is sent to the backend thanks to other unit tests

Reply:
Yeah, this is mainly there to test the Redux logic for the pagination. Again, I went off of the Redux guide to testing: https://redux.js.org/usage/writing-tests to only test complex selectors/reducers that are hard to test elsewhere in the App, hence there's not many tests in the slice test suites